### PR TITLE
NumberInput fixes

### LINKED ISSUE #5928: Format input numbers
- [x] ignore all non-numeric symbols except `,` and `.`
- [x] first used `,` or `.` interpret as a decimal separator
- [x] ignore other usage of `,` or `.`
- [x] format number according the language settings. For example:
    - english → `1,234.56` 
    - czech → `1 234,56`
    - spanish → `1.234,56`
- [x] https://github.com/trezor/trezor-suite/issues/5272
- [x] review validation messages

### PR DESCRIPTION
**Pull request by dahaca**
## Description

* more bug fixes based on [MatthewDavid1984](https://github.com/MatthewDavid1984)'s comments
* remove char code comparisons 

## Related Issue

Resolve https://github.com/trezor/trezor-suite/issues/5928

### CODE REVIEW

File: packages/suite/src/components/suite/NumberInput.tsx [hunk 1]
```diff
@@ -103,41 +111,32 @@ export const NumberInput = ({
                 return newDisplayValue;
             };
 
-            // don't include spaces at the start or the end of a number, for pasting mainly
-            value = value.trim();
-            const cleanValue = cleanValueString(value, locale);
+            let cleanValue = value;
+            if (!isAlreadyClean) {
+                cleanValue = cleanValueString(value, locale);
+            }
```
Comment (komret) [inline packages/suite/src/components/suite/NumberInput.tsx]: I'd prefer cleaning the value outside of this function to keep it simple.

**Review by komret — COMMENTED** [review]
Much better without the charcodes!

**Comment by komret** [comment]
Spanish locale: pasting a decimal number removes the decimal separator.

chrome-capture-2023-3-24

Related question: Do you think it is a good practice to format the number in `handleCopy`? It is convenient for copy-pasting between number inputs, but quite unexpected when you paste the value somewhere else. In an ideal world, the value should be formatted on insertion.

**Comment by komret** [comment]
On the first change in the fiat field, transaction is not composed (it should be as the values are valid). When I add one more digit and the delete it, the transaction is composed:

chrome-capture-2023-3-24 (4)

**Comment by komret** [comment]
These are some minor bugs. Let's discuss if it is worth dealing with them:

- Second bug not fixed: https://github.com/trezor/trezor-suite/issues/5928#issuecomment-1506700164
- Second bug not fixed: https://github.com/trezor/trezor-suite/issues/5928#issuecomment-1506917343
- When you attempt to add a leading zero, the transaction recomposes (not necessary):
chrome-capture-2023-3-24 (1)
- When you attempt to add a second decimal separator, the cursor moves to the end of the value (should stay in place to be consistent with inputting other invalid values):
chrome-capture-2023-3-24 (2)

**Comment by dahaca** [comment]
All fixed except for making separator completely interchangeable, there is not simple solution and I believe we should move on.

File: packages/suite/src/components/suite/NumberInput.tsx (hunk 1 above)
Comment (komret) [reply packages/suite/src/components/suite/NumberInput.tsx]: Sorry, this was not a good  suggestion, I missed that `cleanValue` was used later in the function. Anyway, either solution is ok.

**Comment by komret** [comment]
Looking good except one inconsistency regarding the leading zeroes - this happens when I press `0` (`value` changes to `01.` and I get a weird validation message):

chrome-capture-2023-3-26

**Comment by dahaca** [comment]
@komret  https://github.com/trezor/trezor-suite/pull/8166/commits/81a2173c514e9e91ee5bd3dcc744a423a4f916ef

File: packages/suite/src/components/suite/NumberInput.tsx [hunk 2]
```diff
@@ -21,12 +21,10 @@ const getLocaleSeparators = (locale: Locale) => {
     return { decimalSeparator, thousandsSeparator };
 };
 
-const getLocaleSeparatorCharCodes = (locale: Locale) => ({
-    groupSeparatorCharCode: getLocaleSeparators(locale).thousandsSeparator.charCodeAt(0),
-    decimalsSeparatorCharCode: getLocaleSeparators(locale).decimalSeparator.charCodeAt(0),
-});
+const isValidDecimalString = (value: string) => /^([^.]*)\.[^.]*$/.test(value);
+const hasLeadingZeroes = (value: string) => /^0+(?!\.|$)|(?<=\d\.)0+/g.test(value);
```
Comment (komret) [inline packages/suite/src/components/suite/NumberInput.tsx]: I don't understand why `(?<=\d\.)0+` is there.

Comment (komret) [reply packages/suite/src/components/suite/NumberInput.tsx]: Maybe just rename the variable if you are catching trailing zeroes as well.

**Comment by komret** [comment]
Stumbled over this:

chrome-capture-2023-3-26 (1)

**Comment by komret** [comment]
f4a60f030f46d4a5ec69f2033cf0e64c026574e4 This is an idea how to avoid transaction recomposition when going from `1` to `1.` etc. in send form. Feel free to remove the commit if you have a better way to integrate this.

**Review by komret — APPROVED** [review]
🎖️

File: packages/suite/src/components/suite/NumberInput.tsx (hunk 2 above)
Comment (komret) [reply packages/suite/src/components/suite/NumberInput.tsx]: @dahaca please address this before merging

**Pull request merged**